src/utils/RunSUSHI.js

```diff
@@ -11,6 +13,73 @@ const getRandomPun = utils.getRandomPun;
 const Type = utils.Type;
 const FHIRDefinitions = fhirdefs.FHIRDefinitions;
 
+async function loadAndCleanDatabase(defs, dependencies) {
+  let helperUpdate = await checkForDatabaseUpgrade(dependencies);
+  let loadExternalDependenciesReturn = { defs, emptyDependencies: [] };
+
+  if (helperUpdate.shouldUpdate) {
+    loadExternalDependenciesReturn = await loadExternalDependencies(defs, helperUpdate.version + 1, dependencies);
+    defs = loadExternalDependenciesReturn.defs;
+  } else {
+    loadExternalDependenciesReturn = await loadExternalDependencies(defs, helperUpdate.version, dependencies);
+    defs = loadExternalDependenciesReturn.defs;
+  }
+
+  // Cleans out database of any empty objectStores
+  await cleanDatabase(loadExternalDependenciesReturn.emptyDependencies, helperUpdate.version + 2);
+  return defs;
+}
+
+/**
+ * Run GoFSH
+ * @param {array} input array of JSON definitions to be processed
+ * @param {object} options - config options for GoFSH based on user input and defaults
+ * dependencies: user set, defaults to []
+ * @param {FHIRDefinitions} testDefs - this should only be used by the unit tests so they can provide their own definitions.
+ * @returns {string} the FSH
+ */
+export async function runGoFSH(input, options, testDefs = null) {
+  // Read in the resources, either as JSON objects or as strings
+  const docs = []; // WildFHIR[]
+  input.forEach((resource, i) => {
+    const location = `Input_${i}`;
+    if (typeof resource === 'string') {
+      try {
+        resource = JSON.parse(resource);
+      } catch (e) {
+        logger.error(`Could not parse ${location} to JSON`);
+        return;
+      }
+    }
+    if (gofshUtils.isProcessableContent(resource, location)) {
+      docs.push(new processor.WildFHIR(resource, location));
+    }
+  });
+
+  // Set up the FHIRProcessor
+  const lake = new processor.LakeOfFHIR(docs);
+  let defs = testDefs || new FHIRDefinitions(); // The tests pass in their own set of definitions to use
```

Comment:
*sigh* Yes I completely agree. There should be a way to handle this. Also now that I've responded to your other comments, I feel like we really _should_ have a way to do this within the tests because if we do use `FhirToFsh` in the future, we can't use this approach. However, the problem I was having in the tests was that I can correctly mock out the `loadExternalDependencies` function, which gives back the test set of definitions and updates this variable `defs`. However, when doing that in the test, `fisher.external` isn't updated properly, but it is when the app is run. I'm not sure if maybe I need to somehow mock out `fisher` or `MasterFisher`, or if this is just relying on some object updating by reference that doesn't hold in the tests. If you have any ideas, let me know! I also think something similar might be happening in the tests for running SUSHI, but our test cases are so simple, I didn't notice until now.

Reply:
You have helped me mock the right things and it works! Yay! b2aa777